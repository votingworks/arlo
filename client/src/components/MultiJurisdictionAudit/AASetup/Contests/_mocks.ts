import { IContest } from '../../../../types'

export const contestMocks: {
  [key: string]: { contests: IContest[] }
} = {
  emptyTargeted: {
    contests: [
      {
        id: '',
        name: '',
        isTargeted: true,
        totalBallotsCast: '',
        numWinners: '1',
        votesAllowed: '1',
        jurisdictionIds: [],
        choices: [
          {
            id: '',
            name: '',
            numVotes: '',
          },
          {
            id: '',
            name: '',
            numVotes: '',
          },
        ],
      },
    ],
  },
  emptyOpportunistic: {
    contests: [
      {
        id: '',
        name: '',
        isTargeted: false,
        totalBallotsCast: '',
        numWinners: '1',
        votesAllowed: '1',
        jurisdictionIds: [],
        choices: [
          {
            id: '',
            name: '',
            numVotes: '',
          },
          {
            id: '',
            name: '',
            numVotes: '',
          },
        ],
      },
    ],
  },
  filledTargeted: {
    contests: [
      {
        id: 'contest-id',
        name: 'Contest Name',
        isTargeted: true,
        totalBallotsCast: '30',
        numWinners: '1',
        votesAllowed: '1',
        jurisdictionIds: [],
        choices: [
          {
            id: 'choice-id-1',
            name: 'Choice One',
            numVotes: '10',
          },
          {
            id: 'choice-id-2',
            name: 'Choice Two',
            numVotes: '20',
          },
        ],
      },
    ],
  },
  filledOpportunistic: {
    contests: [
      {
        id: 'contest-id',
        name: 'Contest Name',
        isTargeted: false,
        totalBallotsCast: '30',
        numWinners: '1',
        votesAllowed: '1',
        jurisdictionIds: [],
        choices: [
          {
            id: 'choice-id-3',
            name: 'Choice Three',
            numVotes: '10',
          },
          {
            id: 'choice-id-4',
            name: 'Choice Four',
            numVotes: '20',
          },
        ],
      },
    ],
  },
  filledTargetedWithJurisdictionId: {
    contests: [
      {
        id: 'contest-id',
        name: 'Contest Name',
        isTargeted: true,
        totalBallotsCast: '30',
        numWinners: '1',
        votesAllowed: '1',
        jurisdictionIds: ['jurisdiction-id-1', 'jurisdiction-id-2'],
        choices: [
          {
            id: 'choice-id-1',
            name: 'Choice One',
            numVotes: '10',
          },
          {
            id: 'choice-id-2',
            name: 'Choice Two',
            numVotes: '20',
          },
        ],
      },
    ],
  },
  filledOpportunisticWithJurisdictionId: {
    contests: [
      {
        id: 'contest-id',
        name: 'Contest Name',
        isTargeted: false,
        totalBallotsCast: '30',
        numWinners: '1',
        votesAllowed: '1',
        jurisdictionIds: ['jurisdiction-id-1', 'jurisdiction-id-2'],
        choices: [
          {
            id: 'choice-id-3',
            name: 'Choice Three',
            numVotes: '10',
          },
          {
            id: 'choice-id-4',
            name: 'Choice Four',
            numVotes: '20',
          },
        ],
      },
    ],
  },
  filledTargetedAndOpportunistic: {
    contests: [
      {
        id: 'contest-id',
        name: 'Contest 1',
        isTargeted: true,
        totalBallotsCast: '30',
        numWinners: '1',
        votesAllowed: '1',
        jurisdictionIds: ['jurisdiction-id-1', 'jurisdiction-id-2'],
        choices: [
          {
            id: 'choice-id-1',
            name: 'Choice One',
            numVotes: '10',
          },
          {
            id: 'choice-id-2',
            name: 'Choice Two',
            numVotes: '20',
          },
        ],
      },
      {
        id: 'contest-id',
        name: 'Contest 2',
        isTargeted: false,
        totalBallotsCast: '30',
        numWinners: '1',
        votesAllowed: '1',
        jurisdictionIds: ['jurisdiction-id-1', 'jurisdiction-id-2'],
        choices: [
          {
            id: 'choice-id-3',
            name: 'Choice Three',
            numVotes: '10',
          },
          {
            id: 'choice-id-4',
            name: 'Choice Four',
            numVotes: '20',
          },
        ],
      },
    ],
  },
}

export const contestsInputMocks = {
  inputs: [
    { key: 'Contest Name', value: 'Contest Name' },
    { key: 'Name of Candidate/Choice 1', value: 'Choice One' },
    { key: 'Name of Candidate/Choice 2', value: 'Choice Two' },
    { key: 'Votes for Candidate/Choice 1', value: '10' },
    { key: 'Votes for Candidate/Choice 2', value: '20' },
    { key: 'Total Ballots for Contest', value: '30' },
  ],
  errorInputs: [
    { key: 'Contest Name', value: '', error: 'Required' },
    {
      key: 'Total Ballots for Contest',
      value: '',
      error:
        'Must be greater than or equal to the sum of votes for each candidate/choice',
    },
    {
      key: 'Total Ballots for Contest',
      value: 'test',
      error: 'Must be a number',
    },
    {
      key: 'Total Ballots for Contest',
      value: '-1',
      error: 'Must be a positive number',
    },
    {
      key: 'Total Ballots for Contest',
      value: '0.5',
      error: 'Must be an integer',
    },
    { key: 'Name of Candidate/Choice 1', value: '', error: 'Required' },
    { key: 'Name of Candidate/Choice 2', value: '', error: 'Required' },
    {
      key: 'Votes for Candidate/Choice 1',
      value: '',
      error: 'Required',
    },
    {
      key: 'Votes for Candidate/Choice 1',
      value: 'test',
      error: 'Must be a number',
    },
    {
      key: 'Votes for Candidate/Choice 1',
      value: '-1',
      error: 'Must be a positive number',
    },
    {
      key: 'Votes for Candidate/Choice 1',
      value: '0.5',
      error: 'Must be an integer',
    },
    {
      key: 'Votes for Candidate/Choice 2',
      value: '',
      error: 'Required',
    },
    {
      key: 'Votes for Candidate/Choice 2',
      value: 'test',
      error: 'Must be a number',
    },
    {
      key: 'Votes for Candidate/Choice 2',
      value: '-1',
      error: 'Must be a positive number',
    },
    {
      key: 'Votes for Candidate/Choice 2',
      value: '0.5',
      error: 'Must be an integer',
    },
  ],
}

export default contestsInputMocks
