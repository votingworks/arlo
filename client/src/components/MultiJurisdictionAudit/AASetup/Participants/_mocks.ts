import { readFileSync } from 'fs'
import { join } from 'path'

const jurisdictionFile = new File(
  [
    readFileSync(
      join(
        __dirname,
        '../../../../../public/sample_jurisdiction_filesheet.csv'
      ),
      'utf8'
    ),
  ],
  'jurisdictions.csv',
  { type: 'text/csv' }
)

export default jurisdictionFile
