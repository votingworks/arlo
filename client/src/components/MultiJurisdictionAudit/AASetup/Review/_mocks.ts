import { IAuditSettings } from '../../../../types'

export const settingsMock: {
  [key in 'empty' | 'full' | 'offline' | 'batch']: IAuditSettings
} = {
  empty: {
    state: null,
    electionName: null,
    online: null,
    randomSeed: null,
    riskLimit: null,
    auditType: 'BALLOT_POLLING',
    auditName: 'Test Audit',
  },
  full: {
    state: 'AL',
    electionName: 'Election Name',
    online: true,
    randomSeed: '12345',
    riskLimit: 10,
    auditType: 'BALLOT_POLLING',
    auditName: 'Test Audit',
  },
  offline: {
    state: 'AL',
    electionName: 'Election Name',
    online: false,
    randomSeed: '12345',
    riskLimit: 10,
    auditType: 'BALLOT_POLLING',
    auditName: 'Test Audit',
  },
  batch: {
    state: 'AL',
    electionName: 'Election Name',
    online: false,
    randomSeed: '12345',
    riskLimit: 10,
    auditType: 'BATCH_COMPARISON',
    auditName: 'Test Audit',
  },
}

export const sampleSizeMock = {
  sampleSizes: {
    'contest-id': [
      { prob: 0.54, size: 46, key: 'asn' },
      { prob: 0.7, size: 67, key: '0.7' },
      { prob: 0.5, size: 88, key: '0.5' },
      { prob: 0.9, size: 125, key: '0.9' },
    ],
  },
}

export default settingsMock
