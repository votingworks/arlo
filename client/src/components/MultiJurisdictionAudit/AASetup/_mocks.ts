import { ElementType } from '../../../types'
import { setupStages, stageTitles } from './index'
import { ISidebarMenuItem } from '../../Atoms/Sidebar'

const relativeStages = (
  stage: ElementType<typeof setupStages>,
  state: ISidebarMenuItem['state'] = 'live'
): {
  prevStage: ISidebarMenuItem
  nextStage: ISidebarMenuItem
  menuItems: ISidebarMenuItem[]
} => {
  const prevId = setupStages[setupStages.indexOf(stage) - 1]
  const prevStage: ISidebarMenuItem = {
    id: prevId,
    title: stageTitles[prevId],
    active: false,
    activate: jest.fn(),
    state,
  }
  const nextId = setupStages[setupStages.indexOf(stage) + 1]
  const nextStage: ISidebarMenuItem = {
    id: nextId,
    title: stageTitles[nextId],
    active: false,
    activate: jest.fn(),
    state,
  }
  const menuItems = setupStages.map((s: ElementType<typeof setupStages>) => ({
    id: s,
    title: stageTitles[s],
    active: s === stage,
    activate: jest.fn(),
    state,
  }))
  return { prevStage, nextStage, menuItems }
}

export default relativeStages
