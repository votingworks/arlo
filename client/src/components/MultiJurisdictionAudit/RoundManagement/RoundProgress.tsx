import React from 'react'
import { ProgressBar } from '@blueprintjs/core'
import styled from 'styled-components'
import H2Title from '../../Atoms/H2Title'
import { IRound } from '../useRoundsJurisdictionAdmin'
import { IAuditBoard } from '../useAuditBoards'

const MainBarWrapper = styled.div`
  margin-bottom: 20px;
  width: 500px;
  font-size: 16px;
`

const SmallBarWrapper = styled.div`
  margin-bottom: 10px;
  width: 400px;
`

const RoundProgress = ({
  auditBoards,
  round,
}: {
  auditBoards: IAuditBoard[]
  round: IRound
}) => {
  if (!auditBoards.length) return null
  const sum = (ns: number[]) => ns.reduce((a, b) => a + b)
  const auditedBallots = sum(
    auditBoards.map(ab => ab.currentRoundStatus.numAuditedBallots)
  )
  const sampledBallots = sum(
    auditBoards.map(ab => ab.currentRoundStatus.numSampledBallots)
  )
  return (
    <>
      <H2Title>Round {round.roundNum} Progress</H2Title>
      <MainBarWrapper>
        <span>
          {auditedBallots} of {sampledBallots} ballots audited{' '}
        </span>
        <ProgressBar
          value={auditedBallots / sampledBallots}
          animate={auditedBallots < sampledBallots}
          intent="primary"
        />
      </MainBarWrapper>
      {auditBoards.map(
        ({
          id,
          name,
          currentRoundStatus: { numAuditedBallots, numSampledBallots },
        }) => (
          <SmallBarWrapper key={id}>
            {numSampledBallots > 0 ? (
              <>
                <span>{`${name}: ${numAuditedBallots} of ${numSampledBallots} ballots audited `}</span>
                <ProgressBar
                  value={numAuditedBallots / numSampledBallots}
                  animate={numAuditedBallots < numSampledBallots}
                  intent="primary"
                />
              </>
            ) : (
              <span>{name}: no ballots to audit</span>
            )}
          </SmallBarWrapper>
        )
      )}
    </>
  )
}

export default RoundProgress
