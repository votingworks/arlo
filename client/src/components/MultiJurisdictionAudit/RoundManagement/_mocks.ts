export interface INullResultValues {
  [contestId: string]: {
    [choiceId: string]: null | string | number
  }
}

export const resultsMocks: {
  [key in 'emptyInitial' | 'complete']: INullResultValues
} = {
  emptyInitial: {
    'contest-id-1': {
      'choice-id-1': null,
      'choice-id-2': null,
    },
    'contest-id-2': {
      'choice-id-3': null,
      'choice-id-4': null,
    },
  },
  complete: {
    'contest-id-1': {
      'choice-id-1': 1,
      'choice-id-2': 2,
    },
    'contest-id-2': {
      'choice-id-3': 1,
      'choice-id-4': 2,
    },
  },
}

export default resultsMocks
