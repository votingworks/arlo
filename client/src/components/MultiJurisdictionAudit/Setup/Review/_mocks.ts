export const settingsMock = {
  empty: [
    {
      state: null,
      electionName: null,
      online: null,
      randomSeed: null,
      riskLimit: null,
      auditType: 'BALLOT_POLLING',
    },
    // here for type completion but not used in this context
    /* istanbul ignore next */
    async () => true,
  ],
  full: [
    {
      state: 'AL',
      electionName: 'Election Name',
      online: true,
      randomSeed: '12345',
      riskLimit: 10,
      auditType: 'BALLOT_POLLING',
    },
    // here for type completion but not used in this context
    /* istanbul ignore next */
    async () => true,
  ],
  offline: [
    {
      state: 'AL',
      electionName: 'Election Name',
      online: false,
      randomSeed: '12345',
      riskLimit: 10,
      auditType: 'BALLOT_POLLING',
    },
    // here for type completion but not used in this context
    /* istanbul ignore next */
    async () => true,
  ],
}

export const sampleSizeMock = {
  sampleSizes: {
    'contest-id': [
      { prob: 0.54, size: 46, key: 'asn' },
      { prob: 0.7, size: 67, key: '0.7' },
      { prob: 0.5, size: 88, key: '0.5' },
      { prob: 0.9, size: 125, key: '0.9' },
    ],
  },
}

export default settingsMock
