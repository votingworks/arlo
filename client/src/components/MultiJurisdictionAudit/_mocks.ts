import { contestMocks } from './AASetup/Contests/_mocks'
import { IFileInfo } from './useJurisdictions'
import {
  auditSettings,
  manifestFile,
  talliesFile,
} from './useSetupMenuItems/_mocks'
import { FileProcessingStatus } from './useSetupMenuItems/getJurisdictionFileStatus'
import { IAuditSettings } from '../../types'
import { IBallot } from './RoundManagement/useBallots'
import { IAuditBoard } from './useAuditBoards'
import { IRound } from './useRoundsAuditAdmin'

const manifestFormData: FormData = new FormData()
manifestFormData.append('manifest', manifestFile, manifestFile.name)
const talliesFormData: FormData = new FormData()
talliesFormData.append('batchTallies', talliesFile, talliesFile.name)

export const jaApiCalls = {
  getUser: {
    url: '/api/me',
    response: {
      type: 'jurisdiction_admin',
      name: 'Joe',
      email: 'test@email.org',
      jurisdictions: [
        {
          id: 'jurisdiction-id-1',
          name: 'Jurisdiction One',
          election: {
            id: '1',
            auditName: 'audit one',
            electionName: 'election one',
            state: 'AL',
          },
        },
        {
          id: 'jurisdiction-id-2',
          name: 'Jurisdiction Two',
          election: {
            id: '2',
            auditName: 'audit two',
            electionName: 'election two',
            state: 'AL',
          },
        },
        {
          id: 'jurisdiction-id-3',
          name: 'Jurisdiction Three',
          election: {
            id: '1',
            auditName: 'audit one',
            electionName: 'election one',
            state: 'AL',
          },
        },
      ],
      organizations: [],
    },
  },
  getRounds: {
    url: '/api/election/1/jurisdiction/jurisdiction-id-1/round',
    response: { rounds: [] },
  },
  getBallotManifestFile: (response: IFileInfo) => ({
    url: '/api/election/1/jurisdiction/jurisdiction-id-1/ballot-manifest',
    response,
  }),
  getBatchTalliesFile: (response: IFileInfo) => ({
    url: '/api/election/1/jurisdiction/jurisdiction-id-1/batch-tallies',
    response,
  }),
  getCVRSfile: (response: IFileInfo) => ({
    url: '/api/election/1/jurisdiction/jurisdiction-id-1/cvrs',
    response,
  }),
  getSettings: (response: IAuditSettings) => ({
    url: '/api/election/1/jurisdiction/jurisdiction-id-1/settings',
    response,
  }),
  putManifest: {
    url: '/api/election/1/jurisdiction/jurisdiction-id-1/ballot-manifest',
    options: {
      method: 'PUT',
      body: manifestFormData,
    },
    response: { status: 'ok' },
  },
  putTallies: {
    url: '/api/election/1/jurisdiction/jurisdiction-id-1/batch-tallies',
    options: {
      method: 'PUT',
      body: talliesFormData,
    },
    response: { status: 'ok' },
  },
  getAuditBoards: (auditBoards: IAuditBoard[]) => ({
    url:
      '/api/election/1/jurisdiction/jurisdiction-id-1/round/round-1/audit-board',
    response: { auditBoards },
  }),
  getBallots: (ballots: IBallot[]) => ({
    url: '/api/election/1/jurisdiction/jurisdiction-id-1/round/round-1/ballots',
    response: { ballots },
  }),
}

export const aaApiCalls = {
  getUser: {
    url: '/api/me',
    response: {
      type: 'audit_admin',
      name: 'Joe',
      email: 'test@email.org',
      jurisdictions: [],
      organizations: [
        {
          id: 'org-id',
          name: 'State of California',
          elections: [],
        },
      ],
    },
  },
  getRounds: (rounds: IRound[]) => ({
    url: '/api/election/1/round',
    response: { rounds },
  }),
  getJurisdictions: {
    url: '/api/election/1/jurisdiction',
    response: {
      jurisdictions: [
        {
          id: 'jurisdiction-id-1',
          name: 'Jurisdiction One',
          ballotManifest: { file: null, processing: null },
          currentRoundStatus: null,
        },
        {
          id: 'jurisdiction-id-2',
          name: 'Jurisdiction Two',
          ballotManifest: { file: null, processing: null },
          currentRoundStatus: null,
        },
      ],
    },
  },
  getBatchJurisdictions: {
    url: '/api/election/1/jurisdiction',
    response: {
      jurisdictions: [
        {
          id: 'jurisdiction-id-1',
          name: 'Jurisdiction One',
          ballotManifest: { file: null, processing: null },
          batchTallies: { file: null, processing: null },
          currentRoundStatus: null,
        },
        {
          id: 'jurisdiction-id-2',
          name: 'Jurisdiction Two',
          ballotManifest: { file: null, processing: null },
          batchTallies: { file: null, processing: null },
          currentRoundStatus: null,
        },
      ],
    },
  },
  getJurisdictionFile: {
    url: '/api/election/1/jurisdiction/file',
    response: {
      file: {
        contents: null,
        name: 'file name',
        uploadedAt: 'a long time ago in a galaxy far far away',
      },
      processing: {
        status: FileProcessingStatus.Processed,
        error: null,
        startedAt: 'once upon a time',
        endedAt: 'and they lived happily ever after',
      },
    },
  },
  getContests: {
    url: '/api/election/1/contest',
    response: contestMocks.filledTargeted,
  },
  getSettings: (response: IAuditSettings) => ({
    url: '/api/election/1/settings',
    response,
  }),
  putSettings: {
    url: '/api/election/1/settings',
    options: {
      method: 'PUT',
      body: JSON.stringify(auditSettings.all),
      headers: { 'Content-Type': 'application/json' },
    },
    response: { status: 'ok' },
  },
  getSampleSizes: {
    url: '/api/election/1/sample-sizes',
    response: { sampleSizes: null },
  },
}
