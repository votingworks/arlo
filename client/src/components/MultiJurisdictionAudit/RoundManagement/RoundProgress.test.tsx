import React from 'react'
import { render } from '@testing-library/react'
import RoundProgress from './RoundProgress'
import { auditBoardMocks } from '../useSetupMenuItems/_mocks'

describe('RoundProgress', () => {
  it('renders incomplete round with no audit boards', () => {
    const { container } = render(
      <RoundProgress auditBoards={auditBoardMocks.empty} />
    )
    expect(container).toMatchSnapshot()
  })

  it('renders incomplete round with an audit board', () => {
    const { container } = render(
      <RoundProgress auditBoards={auditBoardMocks.single} />
    )
    expect(container).toMatchSnapshot()
  })

  it('renders incomplete round with two audit boards', () => {
    const { container } = render(
      <RoundProgress auditBoards={auditBoardMocks.double} />
    )
    expect(container).toMatchSnapshot()
  })

  it('renders incomplete round with auditing in progress', () => {
    const { container } = render(
      <RoundProgress auditBoards={auditBoardMocks.started} />
    )
    expect(container).toMatchSnapshot()
  })

  it('renders incomplete round with an audit board with no ballots sampled', () => {
    const { container } = render(
      <RoundProgress auditBoards={auditBoardMocks.noBallots} />
    )
    expect(container).toMatchSnapshot()
  })

  it('renders complete round', () => {
    const { container } = render(
      <RoundProgress auditBoards={auditBoardMocks.signedOff} />
    )
    expect(container).toMatchSnapshot()
  })
})
