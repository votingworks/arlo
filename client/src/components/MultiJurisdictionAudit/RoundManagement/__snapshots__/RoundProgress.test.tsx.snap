// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoundProgress renders complete round 1`] = `
<div>
  <h2
    class="bp3-heading sc-bdVaJa ibVpPq"
  >
    Round 
    1
     Progress
  </h2>
  <div
    class="sc-bwzfXH eeIcAa"
  >
    <span>
      30
       of 
      30
       ballots audited
       
    </span>
    <div
      class="bp3-progress-bar bp3-intent-primary bp3-no-animation"
    >
      <div
        class="bp3-progress-meter"
        style="width: 100%;"
      />
    </div>
  </div>
  <div
    class="sc-htpNat jxYgaV"
  >
    <span>
      Audit Board #01: 30 of 30 ballots audited 
    </span>
    <div
      class="bp3-progress-bar bp3-intent-primary bp3-no-animation"
    >
      <div
        class="bp3-progress-meter"
        style="width: 100%;"
      />
    </div>
  </div>
</div>
`;

exports[`RoundProgress renders incomplete round with an audit board 1`] = `
<div>
  <h2
    class="bp3-heading sc-bdVaJa ibVpPq"
  >
    Round 
    1
     Progress
  </h2>
  <div
    class="sc-bwzfXH eeIcAa"
  >
    <span>
      0
       of 
      30
       ballots audited
       
    </span>
    <div
      class="bp3-progress-bar bp3-intent-primary"
    >
      <div
        class="bp3-progress-meter"
        style="width: 0%;"
      />
    </div>
  </div>
  <div
    class="sc-htpNat jxYgaV"
  >
    <span>
      Audit Board #01: 0 of 30 ballots audited 
    </span>
    <div
      class="bp3-progress-bar bp3-intent-primary"
    >
      <div
        class="bp3-progress-meter"
        style="width: 0%;"
      />
    </div>
  </div>
</div>
`;

exports[`RoundProgress renders incomplete round with an audit board with no ballots sampled 1`] = `
<div>
  <h2
    class="bp3-heading sc-bdVaJa ibVpPq"
  >
    Round 
    1
     Progress
  </h2>
  <div
    class="sc-bwzfXH eeIcAa"
  >
    <span>
      0
       of 
      30
       ballots audited
       
    </span>
    <div
      class="bp3-progress-bar bp3-intent-primary"
    >
      <div
        class="bp3-progress-meter"
        style="width: 0%;"
      />
    </div>
  </div>
  <div
    class="sc-htpNat jxYgaV"
  >
    <span>
      Audit Board #01
      : no ballots to audit
    </span>
  </div>
  <div
    class="sc-htpNat jxYgaV"
  >
    <span>
      Audit Board #02: 0 of 30 ballots audited 
    </span>
    <div
      class="bp3-progress-bar bp3-intent-primary"
    >
      <div
        class="bp3-progress-meter"
        style="width: 0%;"
      />
    </div>
  </div>
</div>
`;

exports[`RoundProgress renders incomplete round with auditing in progress 1`] = `
<div>
  <h2
    class="bp3-heading sc-bdVaJa ibVpPq"
  >
    Round 
    1
     Progress
  </h2>
  <div
    class="sc-bwzfXH eeIcAa"
  >
    <span>
      15
       of 
      30
       ballots audited
       
    </span>
    <div
      class="bp3-progress-bar bp3-intent-primary"
    >
      <div
        class="bp3-progress-meter"
        style="width: 50%;"
      />
    </div>
  </div>
  <div
    class="sc-htpNat jxYgaV"
  >
    <span>
      Audit Board #01: 15 of 30 ballots audited 
    </span>
    <div
      class="bp3-progress-bar bp3-intent-primary"
    >
      <div
        class="bp3-progress-meter"
        style="width: 50%;"
      />
    </div>
  </div>
</div>
`;

exports[`RoundProgress renders incomplete round with no audit boards 1`] = `<div />`;

exports[`RoundProgress renders incomplete round with two audit boards 1`] = `
<div>
  <h2
    class="bp3-heading sc-bdVaJa ibVpPq"
  >
    Round 
    1
     Progress
  </h2>
  <div
    class="sc-bwzfXH eeIcAa"
  >
    <span>
      0
       of 
      60
       ballots audited
       
    </span>
    <div
      class="bp3-progress-bar bp3-intent-primary"
    >
      <div
        class="bp3-progress-meter"
        style="width: 0%;"
      />
    </div>
  </div>
  <div
    class="sc-htpNat jxYgaV"
  >
    <span>
      Audit Board #01: 0 of 30 ballots audited 
    </span>
    <div
      class="bp3-progress-bar bp3-intent-primary"
    >
      <div
        class="bp3-progress-meter"
        style="width: 0%;"
      />
    </div>
  </div>
  <div
    class="sc-htpNat jxYgaV"
  >
    <span>
      Audit Board #02: 0 of 30 ballots audited 
    </span>
    <div
      class="bp3-progress-bar bp3-intent-primary"
    >
      <div
        class="bp3-progress-meter"
        style="width: 0%;"
      />
    </div>
  </div>
</div>
`;
